counterpartylib/lib/blocks.py

```diff
@@ -1200,53 +1206,62 @@ def follow(db):
 
                 # Else: list, parse and save it.
                 else:
-                    try:
-                        with db:
-                            # List the fake block.
-                            cursor.execute('''INSERT INTO blocks(
-                                                block_index,
-                                                block_hash,
-                                                block_time) VALUES(?,?,?)''',
-                                                (config.MEMPOOL_BLOCK_INDEX,
-                                                 config.MEMPOOL_BLOCK_HASH,
-                                                 curr_time)
-                                          )
-
-                            # List transaction.
-                            try:    # Sometimes the transactions can’t be found: `{'code': -5, 'message': 'No information available about transaction'} Is txindex enabled in Bitcoind?`
-                                mempool_tx_index = list_tx(db, None, block_index, curr_time, tx_hash, mempool_tx_index)
-                            except backend.addrindex.BackendRPCError:
-                                raise MempoolError
-
-                            # Parse transaction.
-                            cursor.execute('''SELECT * FROM transactions \
-                                              WHERE tx_hash = ?''',
-                                           (tx_hash,))
-                            transactions = list(cursor)
-                            if transactions:
-                                assert len(transactions) == 1
-                                transaction = transactions[0]
-                                supported = parse_tx(db, transaction)
-                                if not supported:
-                                    not_supported[tx_hash] = ''
-                                    not_supported_sorted.append((block_index, tx_hash))
-                            else:
-                                # If a transaction hasn’t been added to the
-                                # table `transactions`, then it’s not a
-                                # Counterparty transaction.
+                    parse_txs.append(tx_hash)
+
+            # fetch raw for all transactions that need to be parsed
+            # Sometimes the transactions can’t be found: `{'code': -5, 'message': 'No information available about transaction'}`
+            #  - is txindex enabled in Bitcoind?
+            #  - or was there a block found while batch feting the raw txs
+            #  - or was there a double spend for w/e reason accepted into the mempool (replace-by-fee?)
+            try:
+                raw_transactions = backend.getrawtransaction_batch(raw_mempool)
+            except backend.addrindex.BackendRPCError as e:
+                logger.warning('Failed to fetch raw for mempool TXs, restarting loop; %s', (e, ))
+                continue  # restart the follow loop
```

Comment:
sounds impossible to me ...

maybe during bitcoind processing a block, `getrawmempool` returns the mempool prior to the block but the processing of the block is already dropping some of them?

though I doubt that, as far as I can tell, during processing of a block rpc calls are blocked ( I get timeouts every once in a while because the bitcoind on my laptop is still syncing ).